Add some tests that check SDP after transceiver stop

youennf opened the pull request.

youennf committed:
Add some tests that check SDP after transceiver stop

henbos commented on webrtc/RTCRtpTransceiver-stop.html:
```diff
@@ -0,0 +1,60 @@
+  ]);
+}
+
+promise_test(async ()=> {
```
nit: double indent

alvestrand commented on webrtc/RTCRtpTransceiver-stop.html:
```diff
@@ -0,0 +1,60 @@
+
+    const offer = await pc1.createOffer();
+    assert_false(offer.sdp.includes("m=audio"), "offer should not contain an audio m section");
+}, "An offer should be generated even if there is no valid transceiver");
```
This description confuses me, since there is a valid video transceiver.

youennf replied:
Title should be changed

henbos commented on webrtc/RTCRtpTransceiver-stop.html:
```diff
@@ -0,0 +1,60 @@
+
+    const offer = await pc1.createOffer();
+    assert_false(offer.sdp.includes("m=audio"), "offer should not contain an audio m section");
+}, "An offer should be generated even if there is no valid transceiver");
```
Incorrect test name

henbos replied:
Question: does a stopped m= section not show up in the offer?

youennf replied:
Not as per my reading of https://tools.ietf.org/html/draft-ietf-rtcweb-jsep-24#section-5.2
Firefox and Safari behaves accordingly there

alvestrand commented on webrtc/RTCRtpTransceiver-stop.html:
```diff
@@ -0,0 +1,60 @@
+
+    const offer = await pc1.createOffer();
+    assert_true(offer.sdp.includes("a=sendonly"), "m section should be sendonly");
+}, "A stopped sendonly transceiver should generate an inactive m section in the offer");
```
This is testing that a stopped transceiver is present. That seems odd; you should end up with no m=audio section in the SDP.

alvestrand replied:
I see now - this test does negotiation, so it's already generated the section.
This test is not testing for port=0, which is the definition of stopped, though.

youennf replied:
Added an assert for port=0

alvestrand commented:
Needs more review, but very good to have.

youennf commented on webrtc/RTCRtpTransceiver-stop.html:
```diff
@@ -0,0 +1,60 @@
+}
+
+promise_test(async ()=> {
+    const pc1 = new RTCPeerConnection();
```
add clean up

youennf commented:
Updated the tests according review and added some more checks.
As per my reading of https://tools.ietf.org/html/draft-ietf-rtcweb-jsep-24#section-5.2, the first test is correct.
Firefox and Safari behaves the same for all tests except the 3rd one where Firefox always set the m section as inactive.

henbos reviewed (approved):
LGTM
